controllers/common-controller-utils_test.go

```diff
@@ -196,45 +230,70 @@ func expectedRamenSecretData() map[string][]byte {
 
 func TestMirrorPeerSecretReconcile(t *testing.T) {
 	cases := []struct {
+		testCaseID      int
 		name            string
+		ramenNamespace  string
 		manageS3        bool
 		ignoreS3Profile bool
 	}{
 		{
+			testCaseID:      1,
 			name:            "Managing S3 Profile disabled",
+			ramenNamespace:  common.RamenHubNamespace,
 			manageS3:        false,
 			ignoreS3Profile: true,
 		},
 		{
-			name:            "Managing S3 Profile",
+			testCaseID:      2,
+			name:            "Creating new S3 Profile in empty Ramen Config",
+			ramenNamespace:  common.RamenHubNamespace,
+			manageS3:        true,
+			ignoreS3Profile: false,
+		},
+		{
+			testCaseID:      3,
+			name:            "Creating new S3 Profile in filled Ramen Config",
+			ramenNamespace:  "namespace1",
+			manageS3:        true,
+			ignoreS3Profile: false,
+		},
+		{
+			testCaseID:      4,
+			name:            "Updating existing S3 Profile in filled Ramen Config",
+			ramenNamespace:  "namespace2",
 			manageS3:        true,
 			ignoreS3Profile: false,
 		},
 	}
 
+	fakeClient := getFakeClient(t, mgrScheme)
 	for _, c := range cases {
-		fakeClient := getFakeClient(t, mgrScheme)
+		os.Setenv("ODR_NAMESPACE", c.ramenNamespace)
```

Comment:
use t.Setenv()

Reply:
ack

Reply:
t.Setenv() is introduced in 1.17, ODFMO is using 1.16: https://github.com/red-hat-storage/odf-multicluster-orchestrator/blob/main/Dockerfile#L2